src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs

```diff
@@ -348,7 +348,27 @@ private static void AddRestoreSources(List<string> args, string[] sources)
 
         public static XDocument GetRestoreInputFile(string restoreTargetPath, Dictionary<string, string> properties, IEnumerable<string> projectPaths)
         {
+            var commonProjectDirectory = Path.GetDirectoryName(projectPaths.First());
+            foreach (var path in projectPaths)
+            {
+                var directory = Path.GetDirectoryName(path);
+                while (!directory.StartsWith(commonProjectDirectory, StringComparison.OrdinalIgnoreCase))
+                {
+                    commonProjectDirectory = Path.GetDirectoryName(commonProjectDirectory);
+                    if (string.IsNullOrEmpty(commonProjectDirectory))
+                        throw new InvalidOperationException("Projects do not share a common directory.");
+                }
+            }
+
+            var directoryOneLevelUp = Path.GetDirectoryName(commonProjectDirectory);
+
+            // Paths for the Directory.Build.props one level up
+            var buildPropsPath = Path.Combine(directoryOneLevelUp, "Directory.Build.props");
```

Comment:
You don't need to do this. When MSBuild is used to check each project in the solution, the project will import Directory.Build.props itself.